packages/tx/src/tracer.ts

```diff
@@ -303,7 +303,7 @@ export class TxTracer {
 
       const params = {
         query:
-          `tm.event='Tx' and ` +
+          `tm.event='Tx' AND ` +
```

Comment:
Was this the bug???

Reply:
Yes, Osmosis was the only one with this issue. I believe it emerged following the Cosmos SDK upgrade. 

I copied this fix from [Keplr](https://github.com/chainapsis/keplr-wallet/blob/8137c314132f1e7cd1f2cbc5cfecf2a247fc4119/packages/cosmos/src/tx-tracer/index.ts#L342-L361), so it should work reliably for all other chains